prime-router/src/test/kotlin/fhirengine/utils/FhirTranscoderTests.kt

```diff
@@ -1,13 +1,73 @@
 package gov.cdc.prime.router.fhirengine.utils
 
 import assertk.assertThat
+import assertk.assertions.contains
+import assertk.assertions.isNotEmpty
 import assertk.assertions.isNotNull
+import org.hl7.fhir.r4.model.Bundle
+import org.json.JSONException
+import org.json.JSONObject
+import java.util.Date
 import kotlin.test.Test
+import kotlin.test.assertTrue
 
 class FhirTranscoderTests {
     @Test
     fun `test get message engine`() {
         // Just make sure we can get an engine, which means the config is setup correctly.
         assertThat(FhirTranscoder.getMessageEngine()).isNotNull()
     }
+
+    @Test
+    fun `test default parser encode FHIR bundle`() {
+        fun isValidJSONObject(jsonVal: String): Boolean {
+            try {
+                JSONObject(jsonVal)
+            } catch (ex: JSONException) {
+                return false
+            }
+            return true
+        }
+        // Create test FHIR Bundle
+        val testBundle = Bundle()
+        testBundle.type = Bundle.BundleType.MESSAGE
+        testBundle.id = "someid"
+        testBundle.timestamp = Date()
+        testBundle.addEntry()
+
+        val encodedBundle = FhirTranscoder.encode(testBundle)
+
+        assertThat(encodedBundle).isNotEmpty()
+        assertTrue { isValidJSONObject(encodedBundle) }
+
+        // Ensure contains id and type
+        assertThat(encodedBundle).contains("\"id\":\"${testBundle.id}\"")
+        assertThat(encodedBundle).contains("\"type\":\"${testBundle.type.name.lowercase()}\"")
+    }
+
+    @Test
+    fun `test default parser decode FHIR bundle`() {
+        val encodedBundle = """
+            {"resourceType":"Bundle","id":"someid","type":"message","timestamp":"2022-06-14T12:11:50.281-04:00"}
+        """.trimIndent()
+        val decodedBundle = FhirTranscoder.decode(encodedBundle)
+        assertThat(decodedBundle).isNotNull()
+        assertThat(decodedBundle.id == "someid")
+        assertThat(decodedBundle.type == Bundle.BundleType.MESSAGE)
+    }
+    @Test
```

Comment:
Minor: try to be consistent and add a new line before @ Test.  I am not sure if the linter takes care of that.

Reply:
Ohh thats helpful - I assumed the linter would so now I will watch out for it.